packages-web/datagrid/src/components/Pagination.tsx

```diff
@@ -0,0 +1,65 @@
+import { createElement, ReactElement, Dispatch, SetStateAction } from "react";
+
+export interface PaginationProps {
+    page: number;
+    setPage: Dispatch<SetStateAction<number>>;
+    hasMoreItems: boolean;
+}
+
+export interface ClientSidePaginationProps {
+    gotoPage: (page: number) => void;
+    previousPage: () => void;
+    nextPage: () => void;
+    page: number;
+    pageOptions: number[];
```

Comment:
Why not pass a total number of pages instead?

Reply:
Done